src/mqt/bench/evaluation/evaluation_visualization.ipynb

```diff
@@ -0,0 +1,300 @@
+    "eval_data = np.array(eval_data)\n",
+    "\n",
+    "(\n",
+    "    benchmarks,\n",
+    "    num_qubits,\n",
+    "    depth,\n",
+    "    num_all_gates,\n",
+    "    num_nonlocal_gates,\n",
+    "    program_communication,\n",
+    "    critical_depth,\n",
+    "    entanglement_ratio,\n",
+    "    parallelism,\n",
+    "    liveness,\n",
+    ") = [eval_data[:, i] for i in range(len(eval_data[0]))]\n",
+    "perc_multi = num_nonlocal_gates / num_all_gates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0cde4a8b",
+   "metadata": {},
+   "source": [
+    "# Device Distribution"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "588b8449",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "overall_count_indep = 0\n",
+    "indep_qiskit = 0\n",
+    "indep_tket = 0\n",
+    "\n",
+    "overall_count_nat = 0\n",
+    "nat_ibm = 0\n",
+    "nat_rigetti = 0\n",
+    "nat_oqc = 0\n",
+    "nat_ionq = 0\n",
+    "\n",
+    "overall_count_mapped = 0\n",
+    "mapped_ibm_montreal_count = 0\n",
+    "mapped_ibm_washington_count = 0\n",
+    "mapped_oqc_lucy_count = 0\n",
+    "mapped_rigetti_count = 0\n",
+    "mapped_ionq_count = 0\n",
+    "\n",
+    "qiskit_qubit_numbers = []\n",
+    "tket_qubit_numbers = []\n",
+    "\n",
+    "for file in benchmarks:\n",
+    "    str_file = str(file)\n",
+    "    if \"qiskit\" in str_file:\n",
+    "        qiskit_qubit_numbers.append(int(str_file.split(\"_\")[-1].split(\".\")[0]))\n",
+    "    elif \"tket\" in str_file:\n",
+    "        tket_qubit_numbers.append(int(str_file.split(\"_\")[-1].split(\".\")[0]))\n",
+    "\n",
+    "    if \"indep\" in str_file:\n",
+    "        overall_count_indep += 1\n",
+    "        if \"indep_qiskit\" in str_file:\n",
+    "            indep_qiskit += 1\n",
+    "        if \"indep_tket\" in str_file:\n",
+    "            indep_tket += 1\n",
+    "\n",
+    "    if \"nativegates\" in str_file:\n",
+    "        overall_count_nat += 1\n",
+    "        if \"nativegates_ibm\" in str_file:\n",
+    "            nat_ibm += 1\n",
+    "        if \"nativegates_rigetti\" in str_file:\n",
+    "            nat_rigetti += 1\n",
+    "        if \"nativegates_oqc\" in str_file:\n",
+    "            nat_oqc += 1\n",
+    "        if \"nativegates_ionq\" in str_file:\n",
+    "            nat_ionq += 1\n",
+    "    elif \"mapped\" in str_file:\n",
+    "        overall_count_mapped += 1\n",
+    "        if \"mapped_ibm_montreal\" in str_file:\n",
+    "            mapped_ibm_montreal_count += 1\n",
+    "        if \"mapped_ibm_washington\" in str_file:\n",
+    "            mapped_ibm_washington_count += 1\n",
+    "        if \"mapped_oqc_lucy\" in str_file:\n",
+    "            mapped_oqc_lucy_count += 1\n",
+    "        if \"mapped_rigetti\" in str_file:\n",
+    "            mapped_rigetti_count += 1\n",
+    "        if \"mapped_ionq\" in str_file:\n",
+    "            mapped_ionq_count += 1"
```

Comment:
I'd say this "filename" to "counter dictionary" transformation could be packaged as a function in the evaluation module.
I'd put slightly more focus on making the method extensible with future architectures and gate-sets. Maybe TypedDicts or a slightly more sophisticated matching could help here as well.

Reply:
I added two counting methods in the `evaluation.py` file to remove most of the calculations (but simple divisions ) from the jupyter notebook